code/datums/components/acid.dm

```diff
@@ -0,0 +1,178 @@
+/** Component representing acid applied to an object.
+  *
+  * Must be attached to an atom.
+  * Processes, repeatedly damaging whatever it is attached to.
+  * If the parent atom is a turf it applies acid to the contents of the turf.
+  */
+/datum/component/acid
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	/// The strength of the acid on the parent object.
+	var/acid_power
+	/// The volume of acid on the parent object.
+	var/acid_volume
+	/// The ambiant sound of acid eating away at the parent object.
+	var/datum/looping_sound/sizzle
+	/// Used exclusively for melting turfs. TODO: Move integrity to the atom level so that this can be dealt with there.
+	var/parent_integrity = 30
+
+/datum/component/acid/Initialize(_acid_power, _acid_volume)
+	if((_acid_power) <= 0 || (_acid_volume <= 0))
+		return COMPONENT_INCOMPATIBLE // Not enough acid.
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE // Incompatible type. TODO: Rework take_damage to the atom level and move this there.
+	if(isobj(parent))
+		var/obj/O = parent
+		if(O.resistance_flags & UNACIDABLE) // The parent object cannot have acid.
+			return COMPONENT_INCOMPATIBLE
+
+	acid_power = _acid_power
+	set_volume(_acid_volume)
+
+	var/atom/parent_atom = parent
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_OVERLAYS, .proc/on_update_overlays)
+	parent_atom.update_icon()
+	sizzle = new/datum/looping_sound/acid(list(parent), TRUE)
```

Comment:
```suggestion
	sizzle = new /datum/looping_sound/acid(list(parent), TRUE)
```

Reply:
the sound path should be in the var definition and this should just be sizzle = new sizzle(list(parent), TRUE)